worker/uniter/runner/jujuc/secret-get.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package jujuc
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/juju/cmd/v3"
+	"github.com/juju/gnuflag"
+	"gopkg.in/yaml.v2"
+
+	jujucmd "github.com/juju/juju/cmd"
+)
+
+type secretGetCommand struct {
+	cmd.CommandBase
+	ctx Context
+
+	id       string
+	asBase64 bool
+}
+
+// NewSecretGetCommand returns a command to get a secret value.
+func NewSecretGetCommand(ctx Context) (cmd.Command, error) {
+	return &secretGetCommand{ctx: ctx}, nil
+}
+
+// Info implements cmd.Command.
+func (c *secretGetCommand) Info() *cmd.Info {
+	doc := `
+Get the value of a secret with a given secret ID.
+For secrets with a singular value, the decoded string
+is printed, unless --base64 is specified, in which case
+the encoded base64 value is printed.
+Multiple key value secrets are printed as YAML.
+`
+	return jujucmd.Info(&cmd.Info{
+		Name:    "secret-get",
+		Args:    "<ID>",
+		Purpose: "get the value of a secret",
+		Doc:     doc,
+	})
+}
+
+// SetFlags implements cmd.Command.
+func (c *secretGetCommand) SetFlags(f *gnuflag.FlagSet) {
+	f.BoolVar(&c.asBase64, "base64", false,
+		`print the singular secret value as the base64 encoded string`)
+}
+
+// Init implements cmd.Command.
+func (c *secretGetCommand) Init(args []string) error {
+	if len(args) < 1 {
+		return errors.New("missing secret ID")
+	}
+	c.id = args[0]
+	return cmd.CheckEmpty(args[1:])
+}
+
+// Run implements cmd.Command.
+func (c *secretGetCommand) Run(ctx *cmd.Context) error {
+	value, err := c.ctx.GetSecret(c.id)
+	if err != nil {
+		return err
+	}
+
+	var val interface{}
+	if c.asBase64 {
+		// See if the secret is a singular value.
+		if val, err = value.BinaryValue(); err != nil {
+			val = value.BinaryValues()
+		}
+	} else {
+		// See if the secret is a singular value.
+		if val, err = value.StringValue(); err != nil {
+			val, err = value.StringValues()
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	var str string
+	switch v := val.(type) {
+	case string:
+		str = v + "\n"
```

Comment:
The common use case is to just have a single secret blob, eg password or api key etc, that we want printed to stdout directly, not as a key: value. So this behaves similarly to the "smart" formatter we use elsewhere, but it doesn't mess with any scalar values at all. I can add --format here now instead of later.

Reply:
added json and yaml formatters, and this logic goes to "plain"